src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbBuffer.cs

```diff
@@ -661,119 +772,170 @@ internal void ZeroMemory()
 
         internal Guid ReadGuid(int offset)
         {
-            // faster than Marshal.PtrToStructure(offset, typeof(Guid))
-            byte[] buffer = new byte[16];
-            ReadBytes(offset, buffer, 0, 16);
-            return new Guid(buffer);
+            offset += BaseOffset;
+
+            unsafe
+            {
+                Validate(offset, sizeof(Guid));
+            }
+
+            Debug.Assert(0 == offset % ADP.PtrSize, "invalid alignment");
```

Comment:
```suggestion
            Debug.Assert(0 == offset % /* alignof(Guid) /* sizeof(int), "invalid alignment");
```

This is a change is behavior, but `IntPtr` alignment shouldn't be necessary, as `Guid` is aligned to the size of it's largest member, so `int`.